INSTALL.md

```diff
@@ -0,0 +1,315 @@
+[Java Cryptography Extensions](http://www.oracle.com/technetwork/java/javase/downloads/jce8-download-2133166.html).
+
+### Using Maven
+Add the latest java-manta-client dependency to your Maven `pom.xml`.
+
+```xml
+<dependency>
+    <groupId>com.joyent.manta</groupId>
+    <artifactId>java-manta-client</artifactId>
+    <version>LATEST</version>
+</dependency>
+```
+
+Consult [Maven Central search](https://search.maven.org/#search%7Cga%7C1%7Cg%3A%22com.joyent.manta%22%20AND%20a%3A%22java-manta-client%22)
+for a list of all available versions and dependency strings for other project types.
+
+Note: Users are expected to use the same version across sub-packages, e.g. using
+`com.joyent.manta:java-manta-client:3.0.0` with
+`com.joyent.manta:java-manta-client-kryo-serialization:3.1.0` is not supported.
+
+## From Source
+If you prefer to build from source, you'll also need
+[Maven](https://maven.apache.org/), and then invoke:
+
+``` bash
+$ mvn package
+```
+
+Which will compile the jar to `./targets/java-manta-${VERSION}.jar`. You can then
+add it as a dependency to your Java project.
+
+If you want to skip running of the test suite, use the `-DskipTests` property.
+
+# Configuration
+
+Configuration parameters can be set through either system properties or the environment. The following classes allow for
+flexible combination of parameters from different sources:
+
+  - `StandardConfigContext` leaves all values unset and provides named setters
+  - `EnvVarConfigContext` reads values from environment variables (e.g. `export MANTA_URL=value`)
+  - `SystemSettingsConfigContext` reads values from system properties (e.g. from flags `-Dmanta.url=value` or a `.properties` file)
+  - `ChainedConfigContext` allows the combination of multiple `ConfigContext`s
+
+## Parameters
+
+| Default                              | System Property                    | Environment Variable           |
+|--------------------------------------|------------------------------------|--------------------------------|
+| https://us-east.manta.joyent.com:443 | manta.url                          | MANTA_URL                      |
+|                                      | manta.user                         | MANTA_USER                     |
+|                                      | manta.key_id                       | MANTA_KEY_ID                   |
+| $HOME/.ssh/id_rsa                    | manta.key_path                     | MANTA_KEY_PATH                 |
+|                                      | manta.key_content                  | MANTA_KEY_CONTENT              |
+|                                      | manta.password                     | MANTA_PASSWORD                 |
+| 20000                                | manta.timeout                      | MANTA_TIMEOUT                  |
+| 3 (6 for integration tests)          | manta.retries                      | MANTA_HTTP_RETRIES             |
```

Comment:
Is the "6" part actually true? Why?

Reply:
It was true. I don't know if it had changed. It was high to deal with the high failure rate when testing against JPC Manta.

Reply:
I can't find anything in code that would set the higher default. I'll verify the actual number by running a test case and remove the parenthetical if it's no longer different.